### PR TITLE
287 Hardcode the DNS addresses in the code and create a flag for handling multiple DNS

### PR DESCRIPTION
**Pull request by thomas-senechal**
Note that since we only have a DNS deployed on TestNet 17, we can't add default DNS for the others Networks.

### CODE REVIEW

**Comment by SlnPons** [comment]
@thomas-senechal I don't see any comment on the PR, could we add the label "action:merge"?

**Comment by thomas-senechal** [comment]
> @thomas-senechal I don't see any comment on the PR, could we add the label "action:merge"?

Oops.
In fact, I just forgot to put that PR in review. 
We now have to wait for 2 reviews before adding the "action:merge" label.

File: int/api/server.go [hunk 1]
```diff
@@ -54,6 +63,7 @@ func parseNetworkFlag(massaNodeServerPtr *string) {
 		*massaNodeServerPtr = "http://127.0.0.1:33035"
 	}
 
+	os.Setenv(dns.DNSAddressEnvKey, dnsAddressPtr)
```
Comment (gregLibert) [inline int/api/server.go:66]: You should avoid to use the package name in the constant. It's redundant.

File: int/api/websites/all.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ const (
 func RegistryHandler(params operations.AllDomainsGetterParams) middleware.Responder {
 	client := node.NewDefaultClient()
 
-	addressesResult, err := ledger.Addresses(client, []string{dns.DNSRawAddress})
+	addressesResult, err := ledger.Addresses(client, []string{dns.GetDNSAddress()})
```
Comment (gregLibert) [inline int/api/websites/all.go:29]: Same here: Get and DNS are useless.

File: pkg/onchain/dns/dns.go [hunk 3]
```diff
@@ -12,7 +13,11 @@ import (
 	"github.com/massalabs/thyra/pkg/wallet"
 )
 
-const DNSRawAddress = "A15e47ChESAK1SdmGe3b92bybnBvMX2eFaxg23wn3rSdRzFHHGB"
+const DNSAddressEnvKey = "THYRA_DNS_ADDRESS"
+
+func GetDNSAddress() string {
+	return os.Getenv(DNSAddressEnvKey)
+}
```
Comment (gregLibert) [inline pkg/onchain/dns/dns.go:20]:
```suggestion
const EnvKey = "THYRA_DNS_ADDRESS"

func Address() string {
	return os.Getenv(EnvKey)
}
```

**Review by gregLibert — CHANGES_REQUESTED** [review]
:no_good: The readability of the code needs to be improved

File: int/api/server.go [hunk 4]
```diff
@@ -40,20 +42,29 @@ func setAPIFlags(server *restapi.Server, startFlags StartServerFlags) {
...
 
 	parseNetworkFlag(&startFlags.MassaNodeServer)
+
+	if startFlags.DNSAddress != "" {
+		os.Setenv(dns.EnvKey, startFlags.DNSAddress)
+	}
 }
 
 func parseNetworkFlag(massaNodeServerPtr *string) {
+	var dnsAddressPtr string
```
Comment (gregLibert) [inline int/api/server.go:52]: What does Ptr mean ?

**Review by gregLibert — APPROVED** [review]
:thinking:

File: int/api/server.go (hunk 4 above)
Comment (thomas-senechal) [reply int/api/server.go]: Fixed!

**Review by B-Naoufal — APPROVED** [review]

File: int/api/server.go [hunk 5]
```diff
@@ -40,20 +42,29 @@ func setAPIFlags(server *restapi.Server, startFlags StartServerFlags) {
...
+	}
 }
 
 func parseNetworkFlag(massaNodeServerPtr *string) {
+	var dnsAddress string
+
 	switch *massaNodeServerPtr {
 	case "TESTNET":
 		*massaNodeServerPtr = "https://test.massa.net/api/v2"
+		dnsAddress = "A15e47ChESAK1SdmGe3b92bybnBvMX2eFaxg23wn3rSdRzFHHGB"
```
[the remark this answers is not in the document]
Comment (thomas-senechal) [reply int/api/server.go:57]: Discussed in PM

**Pull request merged**